### PR TITLE
Added the @shared webpack alias to the project

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Adding the webpack alias `@shared` like our other existing aliases, e.g. `@utilities`, allows us to avoid writing imports like this

`import { useMediaQuery } from '@shared/useMediaQuery'`

instead of something like this.

`import { useMediaQuery } from '../../shared/components/useMediaQuery'`

This is some pre-work for #12755

## QA Instructions, Screenshots, Recordings

N/A Code that will use this is not part of the PR, so nothing to test aside from the PR going green.

### UI accessibility concerns?

N/A

## Added tests?

- [ ] Yes
- [x] No, and this is why: This is a front-end infrastructure change that isn't in use yet.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [x] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

Sponge Bob sitting ata table saying "Much better"

### CODE REVIEW

**Review by rhymes — APPROVED** [review]
What if we call this `@shared-components` or `@components` ? Mostly because `@shared` might be "shared utilities" or "shared components" or "shared something else". As these are "shared components" I wonder if would be better to just call have something like:

```js
import { useMediaQuery } from '@shared-components/useMediaQuery'
```

or 

```js
import { useMediaQuery } from '@components/useMediaQuery'
```

**Comment by rhymes** [comment]
To argument, better, I think `@components` is my favorite. The reason being that the shared nature of it it's implicit in the `@`. Otherwise we'd have `@shared-utilities` already, instead of `@utilities`. They are all meant to be shared between different parts of the code

File: docs/frontend/webpacker.md [hunk 1]
```diff
@@ -37,6 +37,15 @@ loading split chunks multiple times.
 <%= javascript_packs_with_chunks_tag "webShare", "someOtherPack", defer: true %>
 ```
 
+## Webpack aliases
+
+The project uses
+[webpack aliases](https://webpack.js.org/configuration/resolve/#resolvealias).
+The aliases used in the project can be found in
+https://github.com/forem/forem/blob/master/config/webpack/environment.js#L21-L26
```
Comment (aitchiss) [inline docs/frontend/webpacker.md:45]:
```suggestion
The aliases used in the project can be found under `alias` in
https://github.com/forem/forem/blob/master/config/webpack/environment.js
```

(suggestion / not blocking) Just an idea as the line reference is already going to be out of date once this PR is merged

**Comment by nickytonline** [comment]
> To argument this better: I think `@components` is my favorite. The reason being that the shared nature of it is implicit in the `@`. Otherwise we'd have `@shared-utilities` already, instead of `@utilities`. They are all meant to be shared between different parts of the code

Yeah `@components` does fit better @rhymes. I'll make the change.

**Review by aitchiss — APPROVED** [review]
looks good 👍  one small suggestion for the docs.

also - are we going to update existing references to this folder? We have quite a few places importing those components via relative paths

**Comment by nickytonline** [comment]
> looks good 👍 one small suggestion for the docs.
> 
> also - are we going to update existing references to this folder? We have quite a few places importing those components via relative paths

We'll definitely update those at some point. A 15 minute fix PR for sure. I'm putting this in place because I'm creating a `useMediaQuery` hook and `<MediaQuery />` component for the reading list mobile issue I'm working on.

**Pull request merged**